src/target/riscv/riscv-013.c

```diff
@@ -3809,15 +3847,20 @@ write_memory_abstract(struct target *target, const riscv_mem_access_args_t args)
 {
 	assert(riscv_mem_access_is_write(args));
 
-	RISCV013_INFO(info);
 	int result = ERROR_OK;
-	bool use_aampostincrement = info->has_aampostincrement != YNM_NO;
 
 	/* Convert the size (bytes) to width (bits) */
 	unsigned int width = args.size << 3;
 
-	/* Create the command (physical address, postincrement, write) */
-	uint32_t command = access_memory_command(target, false, width, use_aampostincrement, true);
+	uint32_t command = access_memory_command(target, /* virtual = */ false,
+			width, /* postincrement = */ true, /* is_write = */ true);
+	bool use_aampostincrement = !is_command_unsupported(target, command);
+	if (!use_aampostincrement)
+		/* It is already known that this abstract memory
+		 * access with aampostincrement=1 is not supported.
+		 * So try aampostincrement=0 right away. */
+		command = access_memory_command(target, /* virtual = */ false,
+				width, use_aampostincrement, /* is_write = */ true);
```

Comment:
(same as above)

Unsupported write with aampostincrement=0 could be detected early, too.

You can either do it in this MR or leave a TODO.

```suggestion
	if (!use_aampostincrement) {
		/* It is already known that this abstract memory
		 * access with aampostincrement=1 is not supported.
		 * So try aampostincrement=0 right away. */
		command = access_memory_command(target, /* virtual = */ false,
				width, use_aampostincrement, /* is_write = */ true);
		if (is_command_unsupported(target, command)) {
			LOG_TARGET_DEBUG(target, "Abstract memory write not supported (with or without aampostincrement)")
			return mem_access_result(...);
		 }
	}
```

Reply:
Addressed

Reply:
Same as for read. Please, consider a `TODO` comment instead.

Reply:
Addressed